morph.py

```diff
@@ -7,73 +7,65 @@
-        self.targets.append(line)
-        self.targets.append(line)
+        line = np.dstack((self.std * width / 2 - width / 4, np.zeros(1001)))[0]
+        self.targets.extend([line] * 5)
 
         rad = 100
-        twopi = self.mapping(lambda x: pi*2*x, self.std)
-        xs = self.mapping(lambda t: math.cos(pi*2*t)*rad, self.std)
-        ys = self.mapping(lambda t: math.sin(pi*2*t)*rad, self.std) 
+        twopi = self.mapping(lambda x: pi * 2 * x, self.std)
+        xs = self.mapping(lambda t: math.cos(pi * 2 * t) * rad, self.std)
+        ys = self.mapping(lambda t: math.sin(pi * 2 * t) * rad, self.std)
         circle = np.dstack((xs, ys))[0]
-        self.targets.append(self.interpolate(line,circle,0.25))
-        self.targets.append(self.interpolate(line,circle,0.5))
-        self.targets.append(self.interpolate(line,circle,0.75))   
-        self.targets.append(circle)   
-        self.targets.append(circle)   
-        self.targets.append(circle)   
+        self.targets.append(self.interpolate(line, circle, 0.25))
+        self.targets.append(self.interpolate(line, circle, 0.5))
+        self.targets.append(self.interpolate(line, circle, 0.75))
+        self.targets.extend([circle] * 3)
 
         square = []
-        sideL = 100 # half side length
-        _c_ = math.cos(pi/4)*rad # coordinates of point on the circle where x==y (1st quadrant)
+        sideL = 100  # half side length
+        _c_ = math.cos(pi / 4) * rad  # coordinates of point on the circle where x==y (1st quadrant)
         for p in circle:
             x, y = p
             ax, ay = abs(x), abs(y)
             if ax < ay:
-                fx = self.lerp(-_c_,_c_,-sideL,sideL,ax)*np.sign(x)
-                fy = sideL*np.sign(y)
+                fx = self.lerp(-_c_, _c_, -sideL, sideL, ax) * np.sign(x)
+                fy = sideL * np.sign(y)
             else:
-                fy = self.lerp(-_c_,_c_,-sideL,sideL,ay)*np.sign(y)
-                fx = sideL*np.sign(x)
-            square.append([fx,fy])
-        self.targets.append(self.rotate(np.array(square), pi/4))
-        self.targets.append(self.rotate(np.array(square), pi/4))
-        self.targets.append(self.rotate(np.array(square), pi/4))
-        self.targets.append(self.interpolate(circle, self.rotate(np.array(square), pi/4), 1.6))
-        self.targets.append(self.interpolate(circle, self.rotate(np.array(square), pi/4), 1.6))
-        self.targets.append(self.interpolate(circle, self.rotate(np.array(square), pi/4), 1.6))
-        self.targets.append(self.fuzzify(self.rotate(np.array(square), pi/4), 3,13))
-        self.targets.append(self.fuzzify(self.rotate(np.array(square), pi/4), 3,13))
-        self.targets.append(self.fuzzify(self.rotate(np.array(square), pi/4), 3,13))
+                fy = self.lerp(-_c_, _c_, -sideL, sideL, ay) * np.sign(y)
+                fx = sideL * np.sign(x)
+            square.append([fx, fy])
+        self.targets.extend([self.rotate(np.array(square), pi / 4)] * 3)
```

Comment:
using extend is good but using "+=" is a little but faster and also begineer friendly.